Sources/PackageGraph/Pubgrub/PubgrubDependencyResolver.swift

```diff
@@ -132,10 +132,18 @@ public struct PubgrubDependencyResolver {
 
     /// Execute the resolution algorithm to find a valid assignment of versions.
     public func solve(constraints: [Constraint]) -> Result<[DependencyResolver.Binding], Error> {
-        let root = DependencyResolutionNode.root(package: .root(
-            identity: .plain("<synthesized-root>"),
-            path: .root
-        ))
+        // 👀 is this the right thing to do?
```

Comment:
This seems like an improvement for now, especially for CLI.  For IDEs such as Xcode, where there can be multiple roots and also package dependencies that don't come from packages, I think we'll want to add more information to `PackageGraphRootInput` that can be used in diagnostics.  That should be easy for the IDE to provide and would let you know where the "root pointers" into the package graph are (either a set of root packages or some other external dependency on a package, such as from an iOS app Xcode target to a package).  That could of course be done in a separate PR.

Reply:
also discussed with @neonichu and we agreed this is an improvement